tt_metal/impl/device/mesh_device.hpp

```diff
@@ -4,28 +4,103 @@
 
 #pragma once
 
-#include <memory>
-#include <vector>
 #include <map>
+#include <memory>
 #include <optional>
+#include <vector>
 
+#include "mesh_device_view.hpp"
 #include "tt_metal/impl/device/device.hpp"
 #include "tt_metal/impl/device/mesh_device_view.hpp"
 
 namespace tt::tt_metal {
 
 using DeviceIds = std::vector<int>;
+using MeshDeviceID = std::size_t;
 class MeshDeviceView;
 
-class MeshDevice
-{
-public:
-    MeshShape mesh_shape;
-    std::map<chip_id_t, Device *> managed_devices;
-    std::vector<std::pair<int, Device *>> mesh_devices;
-    std::shared_ptr<MeshDeviceView> view;
+// SystemMesh creates a virtualization over the physical devices in the system.
+// It creates a logical 2D-mesh of devices and manages the mapping between logical and physical device coordinates.
+// It is responsible for the assignment of devices in a MeshDevice to physical devices, and the creation and deletion of
+// device resources.
+class SystemMesh {
+    using LogicalCoordinate = Coordinate;
+    using PhysicalCoordinate = eth_coord_t;
+
+    // Keep track of the devices that were opened so we can close them later. We shouldn't
+    // to keep track of this but DevicePool seems to open all devices associated with an MMIO device id
+    std::unordered_map<MeshDeviceID, std::map<chip_id_t, Device*>> opened_devices;
```

Comment:
it is always good to explicitly write private:

Reply:
done